seacrowd/sea_datasets/snli/snli.py

```diff
@@ -0,0 +1,163 @@
+  journal      = "Data in Brief",
+  volume       = "52",
+  pages        = "109998",
+  publisher    = "Elsevier",
+  doi          = "https://doi.org/10.1016/j.dib.2023.109998",
+  url          = "https://www.sciencedirect.com/science/article/pii/S2352340923010284",
+}
+"""
+
+_DATASETNAME = "snli"
+
+_DESCRIPTION = """\
+The SNLI Indo dataset is derived from the SNLI corpus by translating each premise and hypothesis sentence from English to Indonesia via the Google Cloud Translation API. Premise sentences are crawled image captions from Flickr, and hypothesis sentences are manually created through crowdsourcing. Five annotators are assigned per sentence pair to label the inference relationship as entailment (true), contradiction (false) or neutral (undetermined).
+"""
+
+_HOMEPAGE = "https://data.mendeley.com/datasets/k4tjhzs2gd/1"
+
+_LANGUAGES = ["ind"]
+
+_LICENSE = Licenses.CC_BY_4_0.value
+
+_LOCAL = False
+
+_URLS = {
+    _DATASETNAME: {
+        "train": "https://data.mendeley.com/public-files/datasets/k4tjhzs2gd/files/ee45b2bb-e2ea-47b7-bec4-b6653c467d27/file_downloaded",
+        "val": "https://data.mendeley.com/public-files/datasets/k4tjhzs2gd/files/23aff85c-ff72-48b6-aba1-c1dd5dac216b/file_downloaded",
+        "test": "https://data.mendeley.com/public-files/datasets/k4tjhzs2gd/files/23aff85c-ff72-48b6-aba1-c1dd5dac216b/file_downloaded",
+    }
+}
+
+_SUPPORTED_TASKS = [Tasks.TEXTUAL_ENTAILMENT]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class SNLIDataset(datasets.GeneratorBasedBuilder):
+    """SNLI Indo is derived from the SNLI corpus, where the premise and hypothesis sentences are translated directly from English to Indonesian using the Google Cloud Translation API. This dataset contains ~570k annotated sentence pairs."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="snli_source",
+            version=SOURCE_VERSION,
+            description="snli source schema",
+            schema="source",
+            subset_id="snli",
+        ),
+        SEACrowdConfig(
+            name="snli_seacrowd_pairs",
+            version=SEACROWD_VERSION,
+            description="snli SEACrowd schema",
+            schema="seacrowd_pairs",
+            subset_id="snli",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "snli_source"
+    labels = ["contradiction", "entailment", "neutral"]
+    in_to_eng_labelmap = {
+        "kontradiksi": "contradiction",
+        "keterlibatan": "entailment",
+        "netral": "neutral",
+    }
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "label": datasets.ClassLabel(names=self.labels),
```

Comment:
For the `source` schema, let's keep it as close as possible to the original data format by keeping the label as the original string, i.e., the value of the `in_to_eng_labelmap`.

Reply:
To confirm, so the label should be in bahasa Indonesia, isnt it?

Reply:
Yes, but for accessibility, could we add this comment `# ["contradiction", "entailment", "neutral"` beside the line, @khelli07?

Reply:
I put it beside `labels` since both schema uses Indonesian label